frontend/src/components/Toast.tsx

```diff
@@ -0,0 +1,42 @@
+import { useToast } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+
+type ToastStatus = "error" | "warning" | "success" | "info" | undefined;
+
+type ToastProps = {
+  title: string;
+  message: string;
+  status: ToastStatus;
+};
+
+const ToastHook = (): ((
+  title: string,
+  message: string,
+  status: ToastStatus,
+) => void) => {
+  const [state, setState] = useState<ToastProps | null>();
+  const toast = useToast();
+
+  const addToast = (title: string, message: string, status: ToastStatus) => {
+    const newToast: ToastProps = { title, message, status };
```

Comment:
Since we're only storing one toast as props, do you know what will happen if multiple toasts are emitted at once? (ie does only the latest one get shown, or do they stack). This might be an edge case worth investigating

Reply:
Here is an example (
![image](https://user-images.githubusercontent.com/37082847/172036347-932ce497-e40a-4352-9fe6-4e79b63f1152.png))
the 2nd toast is error just to be able to differentiate between the 2 (in case they overwrote each other). I have also tested navigating to another page while a toast is visible and the toast remains on it's page until it's specified timeout

Reply:
Awesome, thanks for checking!